src/frontend/apps/impress/src/stores/useBroadcastStore.tsx

```diff
@@ -25,26 +38,47 @@ export const useBroadcastStore = create<BroadcastState>((set, get) => ({
     return provider;
   },
   addTask: (taskLabel, action) => {
-    const taskExistAlready = get().tasks[taskLabel];
     const provider = get().getBroadcastProvider();
-    if (taskExistAlready || !provider) {
+    if (!provider) {
+      return;
+    }
+
+    const existingTask = get().tasks[taskLabel];
+    if (existingTask) {
+      existingTask.task.unobserve(existingTask.observer);
+      get().setTask(taskLabel, existingTask.task, action);
       return;
     }
 
     const task = provider.document.getArray<string>(taskLabel);
-    task.observe(() => {
-      action();
-    });
+    get().setTask(taskLabel, task, action);
+  },
+  setTask: (taskLabel: string, task: Y.Array<string>, action: () => void) => {
+    let isInitializing = true;
+    const observer = () => {
+      if (!isInitializing) {
+        action();
+      }
+    };
+
+    task.observe(observer);
+
+    setTimeout(() => {
+      isInitializing = false;
+    }, 1000);
```

Comment:
why ?

Reply:
`task.observe` triggers the task lightly after you bind it to it, it seems like a bug, it is due to the architecture of y.js apparently. We want to exec the task only when we trigger it from broadcast method, so `isInitializing` is a trick to blocked this initial trigger.

https://github.com/numerique-gouv/impress/blob/375418df5438f36f48e8de21697aa650310137c2/src/frontend/apps/impress/src/stores/useBroadcastStore.tsx#L57-L68